query_test.go

```diff
@@ -0,0 +1,104 @@
+package neutrino
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil/gcs"
+	"github.com/btcsuite/btcutil/gcs/builder"
+	"github.com/lightninglabs/neutrino/filterdb"
+	"github.com/lightninglabs/neutrino/lru"
+)
+
+// genRandomBlockHash generates a random block hash using math/rand.
+func genRandomBlockHash() *chainhash.Hash {
+	var seed [32]byte
+	rand.Read(seed[:])
+	hash := chainhash.Hash(seed)
+	return &hash
+}
+
+// getRandFilter generates a random GCS filter that contains numElements. Method
+// taken from filterdb/db_test.go.
+func genRandFilter(numElements uint32) (*gcs.Filter, error) {
+	elements := make([][]byte, numElements)
+	for i := uint32(0); i < numElements; i++ {
+		var elem [20]byte
+		if _, err := rand.Read(elem[:]); err != nil {
+			return nil, err
+		}
+
+		elements[i] = elem[:]
+	}
+
+	var key [16]byte
+	if _, err := rand.Read(key[:]); err != nil {
+		return nil, err
+	}
+
+	filter, err := gcs.BuildGCSFilter(
+		builder.DefaultP, builder.DefaultM, key, elements)
+	if err != nil {
+		return nil, err
+	}
+
+	return filter, nil
+}
+
+// TestFilterCacheStorage verifies the interaction of filters with the
+// lru cache. It verifies that the cache returns the correct filters
+// for a given block and that it evicts older filters.
+func TestFilterCacheStorage(t *testing.T) {
```

Comment:
Good test! 👍

Reply:
This test is a bit outdated since the Size changes, I will rewrite it with additional cases once we agree on the interface for Size, etc.